Improve Installation Instructions

Currently the README installation instructions look like:

```shell
git clone https://github.com/AcademySoftwareFoundation/rmtc.git
cd rmtc
cmake -DCMAKE_INSTALL_PREFIX=/installation/area -B cmake/build -S .
cd cmake/build
make install
source ./rmtc-setup.bash
```
If this is copied and pasted verbatim into a developer's shell, line-for-line, the `make install` step will likely error with:

```shell
% make install
Install the project...
-- Install configuration: ""
-- Installing: /installation/area/RMTC/0.0.0/lib/python-3.9
CMake Error at cmake/cmake_install.cmake:49 (file):
  file INSTALL cannot make directory
  "/installation/area/RMTC/0.0.0/lib/python-3.9": No such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:42 (include)


make: *** [install] Error 1
```

The `README.md` can be improved so that it can be copy/pasted by default so `make install` just works. For example, changing the `cmake` line to this will make it work all the way through:

`cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -B cmake/build -S .`

**Acceptance Criteria:**
* Update the `README.md` Installation `cmake` line so installation commands work out-of-the-box
* Update the documentation to help new developers understand the installation area can be modified
* Update CMakeLists.txt to inform user if the cmake install prefix directory is invalid

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,3 +4,11 @@
 cmake_minimum_required(VERSION 3.27)
 project(RMTC VERSION "0.0.0")
 add_subdirectory(cmake)
+
+include(GNUInstallDirs)
+
+if(NOT EXISTS "${CMAKE_INSTALL_PREFIX}")
+    message(FATAL_ERROR
+        "CMAKE_INSTALL_PREFIX does not exist: ${CMAKE_INSTALL_PREFIX}\n"
+        "Please create it or choose a writable directory.")
+endif()

--- a/README.md
+++ b/README.md
@@ -56,11 +56,13 @@ Assuming you are in your development folder, you can run the following to get go
 ```bash
 git clone https://github.com/AcademySoftwareFoundation/rmtc.git
 cd rmtc
-cmake -DCMAKE_INSTALL_PREFIX=/installation/area -B cmake/build -S .
+cmake -DCMAKE_INSTALL_PREFIX=$HOME/.local -B cmake/build -S .
 cd cmake/build
 make install
 source ./rmtc-setup.bash
 ```
+
+_Note_: You can replace `$HOME/.local` in the cmake command above with your installation directory of choice.
 
 This will clone the repo into your code folder, create build info in cmake/build, then make and install into a given path. The final shell script source sets up the envvars required to execute. 
 

